learning_assistant/utils.py

```diff
@@ -12,7 +12,42 @@
 log = logging.getLogger(__name__)
 
 
-def get_chat_response(message_list):
+def _estimated_message_tokens(message):
+    """
+    Estimates how many tokens are in a given message.
+    """
+    chars_per_token = 3.5
+    json_padding = 8
+
+    return int((len(message) - message.count(' ')) / chars_per_token) + json_padding
+
+
+def get_reduced_message_list(system_list, message_list):
+    """
+    If messages are larger than allotted token amount, return a smaller list of messages.
+    """
+    total_system_tokens = sum(_estimated_message_tokens(system_message['content']) for system_message in system_list)
```

Comment:
This doesn't currently account for if the total_system_tokens is greater than the number of allowed tokens, but I couldn't find anything in https://github.com/2uinc/ai-chatbot-backend on the same codepath as this function that checks for that. I think it's fine as is but probably worth noting in the discovery for https://2u-internal.atlassian.net/browse/MST-2122.